sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/CustomizingStatementHandler.java

```diff
@@ -72,28 +70,40 @@
     private static Stream<BoundCustomizer> parameterCustomizers(Class<?> type, Method method) {
         final Parameter[] parameters = method.getParameters();
 
-        final Function<Integer, Stream<BoundCustomizer>> customizeForParameter =
-                i -> defaultBinding(annotationsFor(parameters[i]))
-                        .map(a -> instantiateFactory(a).createForParameter(a, type, method, parameters[i], i))
-                        .map(c -> (stmt, args) -> c.apply(stmt, args[i]));
-
         return IntStream.range(0, parameters.length)
                 .mapToObj(Integer::valueOf)
-                .flatMap(customizeForParameter);
+                .flatMap(i -> eachParameterCustomizers(type, method, parameters[i], i));
     }
 
-    /** Add a {@code @Bind} annotation to any parameter with no annotations. */
-    private static Stream<Annotation> defaultBinding(Stream<Annotation> annotations) {
-        final List<Annotation> a = annotations.collect(Collectors.toList());
-        if (!a.isEmpty()) {
-            return a.stream();
+    private static Stream<BoundCustomizer> eachParameterCustomizers(Class<?> type, Method method, Parameter parameter, Integer i) {
+
+        List<BoundCustomizer> customizers = annotationsFor(parameter)
+                .map(a -> instantiateFactory(a).createForParameter(a, type, method, parameter, i))
+                .<BoundCustomizer>map(c -> (stmt, args) -> c.apply(stmt, args[i])).collect(Collectors.toList());
+
+        if (!customizers.isEmpty()) {
+            return customizers.stream();
         }
-        return Stream.of(Bind.DEFAULT);
+
+        return Stream.of(defaultParameterCustomizer(type, method, parameter, i));
+    }
+
+    /**
+     * Default parameter customizer for parameters with no annotations.
+     */
+    private static BoundCustomizer defaultParameterCustomizer(Class<?> type, Method method, Parameter parameter, Integer i) {
+        return (stmt, args) -> getDefaultParameterCustomizerFactory(stmt)
+                .createForParameter(null, type, method, parameter, i)
+                .apply(stmt, args[i]);
```

Comment:
I'd rather we introduce a separate factory interface for the default customizer, similar to what #754 is doing with default handlers. It feels weird to pass a `null` for the annotation, when the use case is specifically for when there _is_ no annotation. e.g.

```java
interface DefaultParameterCustomizerFactory {
    SqlStatementParameterCustomizer createForParameter(Class<?> sqlObjectType,
                                                       Method method,
                                                       Parameter param,
                                                       int index);
}
```